addons/mail/static/src/messaging/entity/entity/entity.js

```diff
@@ -2,28 +2,45 @@ odoo.define('mail.messaging.entity.Entity', function (require) {
 'use strict';
 
 const {
+    fields: {
+        many2one,
+    },
     registerNewEntity,
 } = require('mail.messaging.entity.core');
 
-const STORED_RELATION_PREFIX = `_`;
-
 function EntityFactory() {
 
     class Entity {
 
+        /**
+         * @param {mail.messaging.messagin_env} env
+         * @param {any} data
+         */
+        constructor(env, data) {
```

Comment:
comment that create must be called, throw here if not called from create maybe

move code into create